### PR TITLE
group flaky tests and run them exclusively 

### PR DESCRIPTION
**Pull request by TriplEight**
cc https://github.com/paritytech/ci_cd/issues/758
cc https://github.com/paritytech/substrate/issues/11321

### CODE REVIEW

**Review by sergejparity — APPROVED** [review]

File: scripts/ci/gitlab/nextest.toml [hunk 1]
```diff
@@ -108,17 +108,25 @@ store-success-output = false
...
 [[profile.default.overrides]]
-filter = 'test(/(^ui$|_ui|ui_)/)'
-test-group = 'serial-integration'
+# To list all the tests, use this command:
+# cargo nextest list --workspace --locked --release --features=runtime-benchmarks
+#
+# To check if you added your test to a filter and group correctly, use this command:
+# cargo nextest show-config test-groups --config-file scripts/ci/gitlab/nextest.toml --features runtime-benchmarks --workspace --locked --release --manifest-path ./bin/node/cli/Cargo.toml
+# For the non-default [[profile.default.overrides]] i.e. [[profile.local.overrides]] add `--profile local` to the above command
+filter = "test(follow_report_multiple_pruned_block) | test (follow_forks_pruned_block) | test(can_sync_small_non_best_forks) | test(telemetry_works) | test(running_the_node_works_and_can_be_interrupted) | test(notifications_back_pressure)"
```
Comment (oleg-plakida) [inline scripts/ci/gitlab/nextest.toml:123]: Why did you change the filter? It will not work for `ui` type of tests.

File: scripts/ci/gitlab/nextest.toml [hunk 2]
```diff
@@ -108,17 +108,25 @@ store-success-output = false
...
+# Flaky tests are executed exclusively
+test-group = 'flaky'
```
Comment (oleg-plakida) [inline scripts/ci/gitlab/nextest.toml:125]: These name is misleading. Flaky tests are differ from mutual exclusive ones.

**Review by oleg-plakida — CHANGES_REQUESTED** [review]

File: scripts/ci/gitlab/nextest.toml (hunk 2 above)
Comment (TriplEight) [reply scripts/ci/gitlab/nextest.toml:125]: this group is sorta TODO by design, refers to the tests that are noted to be flaky and require fixing, but not yet.

Comment (oleg-plakida) [reply scripts/ci/gitlab/nextest.toml:125]: Initialy group was created for test which require single thread run. They are not flaky. We know the nature and behavior of them is consistent.

File: scripts/ci/gitlab/nextest.toml [hunk 3]
```diff
@@ -120,6 +120,10 @@ store-failure-output = true
 # To check if you added your test to a filter and group correctly, use this command:
 # cargo nextest show-config test-groups --config-file scripts/ci/gitlab/nextest.toml --features runtime-benchmarks --workspace --locked --release --manifest-path ./bin/node/cli/Cargo.toml
 # For the non-default [[profile.default.overrides]] i.e. [[profile.local.overrides]] add `--profile local` to the above command
+filter = 'test(/(^ui$|_ui|ui_)/)'
```
Comment (oleg-plakida) [inline scripts/ci/gitlab/nextest.toml:123]: That won't work. `[[profile.default.overrides]]` means that last one will override the prev one. If you need to add another group you have to create another profile. Or mb there is another way. Check docs for nextest.

**Review by oleg-plakida — CHANGES_REQUESTED** [review]

File: scripts/ci/gitlab/nextest.toml (hunk 3 above)
Comment (oleg-plakida) [reply scripts/ci/gitlab/nextest.toml:123]: It does. My bad.

**Review by oleg-plakida — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2813512
